llvm/lib/Support/SpecialCaseList.cpp

```diff
@@ -63,6 +63,11 @@ Error SpecialCaseList::Matcher::insert(StringRef Pattern, unsigned LineNumber,
                        .moveInto(Pair.first))
       return Err;
     Pair.second = LineNumber;
+  } else {
```

Comment:
Please move llvm/lib/Support/SpecialCaseList.cpp and llvm/unittests/Support/SpecialCaseListTest.cpp into a separate list

Reply:
Also we need a patch to Matcher::Globs -> vector

As I see we don't use Map property, but it will mess up the order.

Reply:
And "Pair.second = LineNumber;" is not a solution:
More generic case it will solve https://github.com/llvm/llvm-project/pull/140855 (please review and merge)

Reply:
I believe the change inside `SpecialCaseList::Matcher::insert` is needed. 
Consider the following example,
```
1 src:*
2 src:*/test1.c
3 src:*/test1.c=sanitize
4 src:*/test1.c
```
`src:*/test1.c` at line 2 and `src:*/test1.c` at line 4 shares the same Section, Prefix and Query and Catogory. As a result, when the previous implementation will skip updating the lineNo. Without the change, we can not know the correct line number even if we do a reverse order iteration. (I tested it by removing the change inside `SpecialCaseList::Matcher::insert` and walking Globs/Regex in reverse order. The test will fail for src.ignorelist.contradict4

> Also we need a patch to Matcher::Globs -> vector

Will change it in a new PR.

Reply:
> implementation without this PR will skip updating the lineNo

It will not be needed after Matcher::Globs -> vector change

Reply:
> It will not be needed after Matcher::Globs -> vector change

Yeah, https://github.com/llvm/llvm-project/pull/140964 is landed. The change is easy now